### PR TITLE
use bazel wrapper for "bazel" invocations in grpc workspace

### PR DESCRIPTION
**Pull request by jtattermusch**
Followup for https://github.com/grpc/grpc/pull/19127

Use the trick where //tools/bazel hijacks "bazel" invocations in our workspace.

### CODE REVIEW

**Comment by jtattermusch** [comment]
Two more things to solve:
- avoid using wrapper in bazel wrapper in docker images where the right version of bazel is already installed
- allow some way to disable the "//tools/bazel" hijacking.

File: tools/bazel [hunk 1]
```diff
@@ -19,35 +19,39 @@
...
+# script can be used to hijack "bazel" invocations in given workspace.
 
 set -e
 
 VERSION=0.24.1
 
-CWD=`pwd`
+echo "INFO: Running bazel wrapper (see //tools/bazel for details), bazel version $VERSION will be used instead of system-wide bazel installation."
+
+CWD=$(pwd)
```
Comment (gnossen) [inline tools/bazel]: Nit: This doesn't look ideal. Could we use a subshell with a single `cd` at the beginning? Or `pushd` and `popd`?

**Comment by jtattermusch** [comment]
> Two more things to solve:
> 
> * avoid using wrapper in bazel wrapper in docker images where the right version of bazel is already installed
> * allow some way to disable the "//tools/bazel" hijacking.

Hopefully I resolved both of these. PTAL.

File: tools/bazel [hunk 2]
```diff
@@ -24,13 +24,20 @@
 
 set -e
 
+# First of all, if DISABLE_BAZEL_WRAPPER is set, just use BAZEL_REAL as set by
+# https://github.com/bazelbuild/bazel/blob/master/scripts/packages/bazel.sh
+# that originally invoked this script.
+if [ "${BAZEL_REAL}" != "" ] && [ "${DISABLE_BAZEL_WRAPPER}" != "" ]
```
Comment (gnossen) [inline tools/bazel]: Nit: `if [ -n ... ]`?

Comment (jtattermusch) [reply tools/bazel]: Is there really a difference?

**Comment by jtattermusch** [comment]
Known failures:
https://github.com/grpc/grpc/issues/18416

**Pull request merged**